src/type_translator.ts

```diff
@@ -624,6 +624,7 @@ export class TypeTranslator {
       }
       return typeStr;
     } else if (type.objectFlags & ts.ObjectFlags.Anonymous) {
+      this.seenAnonymousTypes.add(type);
```

Comment:
It feels like this belongs in translateAnonymousType.  Maybe the rest of the code in this block (before the call to this.translateAnonymousType) belongs in translateAnonymousType as well?

Reply:
Fair enough.  After doing so, it made it more obvious that type.symbol was being checked redundantly, so I simplified slightly.

Reply:
Looks great.